### ISSUE TITLE
Training on custom dataset

### ISSUE THREAD

**Issue by crepejung00** [nianticlabs/mickey#18]
Hi, thanks for the great work!

I am trying to train Mickey on custom large-scale datasets, such as RealEstate10K. While I have successfully implemented the dataloader for Mickey, I am not sure if the training is being done correctly just by seeing the loss graphs or the depth map shown during training. Specifically, I am not seeing any monotonic decrease in the error metrics during training which is shown in the screen shot below.
![image](https://github.com/user-attachments/assets/cd49f4b4-abac-4129-a9b4-0f608437c3ff)

I have tried training Mickey on the sample dataset of MapFree, but this case also showed a similar training signal. Are there any signals I can verify if my training is being done correctly?

Thanks!

**Comment by axelBarroso** [comment]
Hello, thanks for your interest in our work!

What loss function are you using? If using the VCRE (with soft clipping) the loss should start at 1.0 and decrease to 0.3-0.2 by the end of the training. In the [config file](https://github.com/nianticlabs/mickey/blob/main/config/MicKey/overlap_score.yaml), you should see under the `LOSS_CLASS` section: 
`LOSS_FUNCTION: "VCRE"`
`SOFT_CLIPPING: True`

One problem with the REINFORCE pipeline in MicKey is that it is slow to train, we have just pushed a new RANSAC that speeds up the training pipeline. Besides, we have added two configurations ( _curriculum_learning_warm_up.yaml_ and _overlap_score_warm_up.yaml_), they use low-resolution images and do not include the null hypothesis (both changes should improve the converge speed). 
 
 As a reference, see the training curves when using _overlap_score_warm_up.yaml_:
![Screenshot 2024-09-19 at 09 50 17](https://github.com/user-attachments/assets/dde33c3b-6c6c-477c-87a2-f43eb3d66544)

I would suggest using the latest version, and see whether you can replicate the training curves for Map-free. I am happy to help debug why RealEstate10K is not working after this sanity check.

Thanks!

**Comment by crepejung00** [comment]
Hi, thanks for the quick reply!
I have tried the latest version, but the loss value starts from a very small value from the beginning(around 1e-3) while the loss_rot is about 0.3 and loss_trans is about 1.5. Could this be due to the large batch sizes? Instead of using 4 GPUs I am training with 1 using the batch_size * 4 shown in the configuration file.

**Comment by axelBarroso** [comment]
Hi, thanks for trying the new version that fast!

Are you training with the map-free data? You could use this [config file](https://github.com/nianticlabs/mickey/blob/main/config/MicKey/overlap_score_warm_up.yaml) for now, to verify that the pipeline works as expected. Set to 1 the GPU (the training curves I shared above are using only 1 GPU and batch size 24). 

The tensorboard also stores keypoints, depths, and matches. Are you able to visualise any of those to see whether the network is generating something meaningful? If the loss is that low, I would expect something like the figure below:
![Screenshot 2024-09-19 at 15 46 41](https://github.com/user-attachments/assets/9b8fd168-150d-4ea8-9a1f-d9f73a154a3c)

**Comment by crepejung00** [comment]
I am actually trying out with the realestate10k dataset, but I figure that this maybe due to the image pairs that are sampled have large overlaps making the problem too easy. The correspondence is not perfect but near correct from the beginning while the depth is not being learned.
The visualizations of the early steps of training are shown below.
<img width="1646" alt="image" src="https://github.com/user-attachments/assets/1facb727-3c96-4412-8d45-f8dd40076441">
<img width="2238" alt="image" src="https://github.com/user-attachments/assets/5305d277-c983-4b62-a2d6-1a43209080e7">

Thank you very much for the help!

**Comment by axelBarroso** [comment]
Hello, I think I know what the problem is, MicKey expects a metric dataset to train. As far as I know (do please correct me), RealEstate10K is an unscaled dataset, and then it cannot be trained with our REINFORCE pipeline out of the box. 

One option could be to switch to pose supervision in the config file and then only use the rotation loss (not the translation). Hope this helps, do please reopen the issue if you have further questions.

**Comment by crepejung00** [comment]
Hi, thank you very much for the kind explanation!
I do think that could be the main problem! I will try only with the rotation loss.
Thank you again for the great work and kind help!

Best regards,
Jaewoo.
